src/strategyHelpers/blueGreen/deploy.ts

```diff
@@ -0,0 +1,139 @@
+import * as core from '@actions/core'
+
+import {Kubectl} from '../../types/kubectl'
+import {RouteStrategy} from '../../types/routeStrategy'
+
+import {
+   BlueGreenDeployment,
+   BlueGreenManifests,
+   deployWithLabel,
+   getManifestObjects,
+   GREEN_LABEL_VALUE,
+   deployObjects
+} from './blueGreenHelper'
+import {setupSMI} from './smiBlueGreenHelper'
+
+import {routeBlueGreenForDeploy} from './route'
+
+export async function deployBlueGreen(
+   kubectl: Kubectl,
+   files: string[],
+   routeStrategy: RouteStrategy
+): Promise<BlueGreenDeployment> {
+   let blueGreenDeployment: BlueGreenDeployment
```

Comment:
this pattern shows up a lot in the codebase where we create a variable then set it

this is better
```javascript
const blueGreenDeployment = await (async () => {
   switch (routeStrategy) {
      case RouteStrategy.INGRESS:
         return await deployBlueGreenIngress(kubectl, files)
      case RouteStrategy.SMI:
         return await deployBlueGreenSMI(kubectl, files)
      default:
         return deployBlueGreenService(kubectl, files)
   }
})()
```

It's more explicit what the variable will be plus you can use a const for a more functional style to avoid state bugs.